src/Sonarr.Api.V3/Indexers/ReleasePushModule.cs

```diff
@@ -1,50 +1,111 @@
+            _releaseValidator.RuleFor(s => s.PublishDate).NotEmpty();
 
             Post["/push"] = x => ProcessRelease(this.Bind<ReleaseResource>());
-
-            PostValidator.RuleFor(s => s.Title).NotEmpty();
-            PostValidator.RuleFor(s => s.DownloadUrl).NotEmpty();
-            PostValidator.RuleFor(s => s.DownloadProtocol).NotEmpty();
-            PostValidator.RuleFor(s => s.PublishDate).NotEmpty();
         }
 
         private Response ProcessRelease(ReleaseResource release)
         {
+            var validationFailures = _releaseValidator.Validate(release).Errors;
+
+            if (validationFailures.Any())
+            {
+                throw new ValidationException(validationFailures);
+            }
+
             _logger.Info("Release pushed: {0} - {1}", release.Title, release.DownloadUrl);
 
             var info = release.ToModel();
 
             info.Guid = "PUSH-" + info.DownloadUrl;
 
+            ResolveIndexer(info);
+
             var decisions = _downloadDecisionMaker.GetRssDecision(new List<ReleaseInfo> { info });
             _downloadDecisionProcessor.ProcessDecisions(decisions);
 
-            return MapDecisions(decisions).First().AsResponse();
+            var firstDecision = decisions.FirstOrDefault();
+
+            if (firstDecision?.RemoteEpisode.ParsedEpisodeInfo == null)
+            {
+                throw new ValidationException(new List<ValidationFailure>{ new ValidationFailure("Title", "Unable to parse", release.Title) });
+            }
+
+            return MapDecisions(new [] { firstDecision }).AsResponse();
+        }
+
+        private void ResolveIndexer(ReleaseInfo release)
+        {
+            if (release.IndexerId == 0 && release.Indexer.IsNotNullOrWhiteSpace())
+            {
+                var indexer = _indexerFactory.All().FirstOrDefault(v => v.Name == release.Indexer);
+                if (indexer != null)
+                {
+                    release.IndexerId = indexer.Id;
+                    _logger.Debug("Push Release {0} associated with indexer {1} - {2}.", release.Title, release.IndexerId, release.Indexer);
+                }
+                else
+                {
+                    _logger.Debug("Push Release {0} not associated with unknown indexer {1}.", release.Title, release.Indexer);
+                }
+            }
+            else if (release.IndexerId != 0 && release.Indexer.IsNullOrWhiteSpace())
+            {
+                try
+                {
+                    var indexer = _indexerFactory.Get(release.IndexerId);
+                    release.Indexer = indexer.Name;
+                    _logger.Debug("Push Release {0} associated with indexer {1} - {2}.", release.Title, release.IndexerId, release.Indexer);
+                }
+                catch (ModelNotFoundException)
+                {
+                    _logger.Debug("Push Release {0} not associated with unknown indexer {0}.", release.Title, release.IndexerId);
```

Comment:
'not' 'unknown', sounds like a double negative.

Reply:
Copy paste from the v2 implementation, I'll fix both (also didn't intend for this commit to land here, oops).